### DISCUSSION TITLE
Chart titles in Plot or Plotlyjs?

### DISCUSSION THREAD

**Discussion by gganapat** [WLJSTeam/wljs-notebook#138] [General]
Hi:

Notebook 2.0 is a huge improvement over earlier versions which was quite buggy. It's becoming more feasible to do relatively complex computations. The documentation progress is commendable. I have a question on plotting. I could not get layout to work with Plotly. I was assuming something like this would work.
```
trace1 = <|
	"x" -> dat, 
	"y" -> power, 
	"mode" -> "markers",
    "layout" -> <|"xaxis" -><|"title" -> "Date"|>, "yaxis" -><|"title" -> "Power"|>|>;
|>;

Plotly[trace1]
```
Plot without layout works but I tried different layout options and none work. Trying to add Plot Title, change width, height also failed. Any tips?

Thanks!
Gani -

**Comment by JerryI** [comment]
We are very glad to hear that! 🚀

Regarding `Plotly` it was actually a bug.., that we did not pass `Layout` correctly. A new version is uploaded, you only need to open `Settings` and click `Check Updates` it will force frontend to fetch new version on the next startup.

The correct example then will look like

```mathematica
trace1 = <|
	"x" -> Range[100], 
	"y" -> Sinc @ Range[100] // N, 
	"mode" -> "markers"
|>;

layout = <|"xaxis" -><|"title" -> "Date"|>, "yaxis" -><|"title" -> "Power"|>|>;

Plotly[trace1, layout]
```

<img width="555" alt="Screenshot 2024-05-03 at 00 01 32" src="https://github.com/JerryI/wolfram-js-frontend/assets/4111822/1cab8d38-0253-495e-8a9a-37e32cbf46a4">

*probably we need to use different margin parameter for default, but you can specify it in layout association*

For more information in API, please check 
- [Plotly](https://jerryi.github.io/wljs-docs/frontend/Reference/Plotly/)
- PlotlyAddTraces
- PlotlyDeleteTraces
- PlotlyExtendTraces
- ... etc in the same section


### Regarding native `Plot` function
Yes, it still does not support `PlotTitle`, but `PlotLegends` kinda work in most cases.

**Reply by gganapat** [reply]
Thanks! That worked.
